PrestaShop/PrestaShop
# POC - ICU translation test

| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | develop
| Description?      | A test of translations using the ICU syntax (https://unicode-org.github.io/icu/userguide/format_parse/messages/). This will finally allow us to properly translate plural forms in all languages.
| Type?             | new feature
| Category?         | CO
| BC breaks?        | no
| Deprecations?     | no
| How to test?      | See below
| Fixed ticket?     | Fixes #15870
| Related PRs       | 
| Sponsor company   | 

### How to test
- Install prestashop in czech, checkout to this PR so you have translations there.
- Delete `var/cache/prod|dev/translations`.
- See any product in FO.

### Findings
- Symfony uses the domain ending suffix to detect if it should use IntlFormatter or MessageFormatter. If we manage to change this, it's all we must do. By temporarily removing this condition, everything works:
  - We can use the normal catalog.
  - Translation shows up in the interface.
  - We can modify the translation and use SQL version of it.

### What now
- Do we just translate everything like it is ICU format?
- Do we keep adding ICU to domain like it should be done, but load our messages from normal domains? It would make it safer.
